### ISSUE TITLE
Object syntax - Lenient property formatting.

### ISSUE THREAD

**Issue by jordanjlatimer** [emotion-js/emotion#2270]
**Description:**
Version: 11.1.5

I've discovered that when using object syntax, you can be more relaxed when specifying child elements or pseudo elements than the documentation seems to let on.

The doc's examples do this:
```
// pseudos
{
  "&:hover": {...}
}

//children
{
  "& div": {...}
}
```

But I've found the following also work:
```
//pseudos
{
  ":hover": {...}
}

//children
{
  " div": {...},
  "div": {...},
  div: {...},
}
```
Is this behavior intentional? Should one avoid using the additional formats I showed?

**Documentation links:**
https://emotion.sh/docs/object-styles

**Comment by Andarist** [comment]
The object styles are serialized to a string so in a case like `{ " div": {} }` the whitespace doesn't matter - in the very same way it doesn't matter in a string style.

As to `&:hover` vs `:hover` - I would encourage you to use the former because `:hover` should actually be equivalent to `& :hover`, we are only patching to be equivalent to `&:hover`. And the patching requires some additional, small, runtime cost. This comes from the fact that we were previously using Stylis 3 which has chosen incorrect behavior for this case if we consider what has been in Sass, Less, and others the correct one. This has been fixed in Stylis 4 but to maintain compatibility on our side we are patching things at runtime so people can migrate more easily.

**Comment by oliviertassinari** [comment]
> This has been fixed in Stylis 4 but to maintain compatibility on our side we are patching things at runtime so people can migrate more easily.

I think that it could be good to eventually remove the patch for `:hover`. I see issues with this:

1. It can create confusion: https://github.com/mui/material-ui/issues/30660#issuecomment-1368476632 with the other selectors.
2. The spec for https://www.w3.org/TR/css-nesting-1/ seems to require either `&` or `:nest` to be used.
3. It breaks what people learn with SASS and LESS. We can compare the output of 

```css
.id {
  color: hotpink;

  :hover {
    color: red;
  }
}
```

- [LESS](https://lesscss.org/less-preview/#eyJjb2RlIjoiI2xpYigpIHtcbiAgICAuY29sb3JzKCkge1xuICAgICAgQHByaW1hcnk6IGJsdWU7XG4gICAgICBAc2Vjb25kYXJ5OiBncmVlbjtcbiAgICB9XG4gICAgLnJ1bGVzKEBzaXplKSB7XG4gICAgICBib3JkZXI6IEBzaXplIHNvbGlkIHdoaXRlO1xuICAgIH1cbiAgfVxuICBcbiAgLmJveCB3aGVuICgjbGliLmNvbG9yc1tAcHJpbWFyeV0gPSBibHVlKSB7XG4gICAgd2lkdGg6IDEwMHB4O1xuICAgIGhlaWdodDogKCR3aWR0aCAvIDIpO1xuICB9XG4gIFxuICAuYmFyOmV4dGVuZCguYm94KSB7XG4gICAgQG1lZGlhIChtaW4td2lkdGg6IDYwMHB4KSB7XG4gICAgICB3aWR0aDogMjAwcHg7XG4gICAgICAjbGliLnJ1bGVzKDFweCk7XG4gICAgfVxuICB9IiwiYWN0aXZlVmVyc2lvbiI6IjQueCJ9) ⬇️

```css
.id {
  color: hotpink;
}
.id :hover {
  color: red;
}
```

- [SASS](https://www.sassmeister.com/) ⬇️

```css
.id {
  color: hotpink;
}
.id :hover {
  color: red;
}
```

- [Emotion](https://codesandbox.io/s/twilight-leaf-f8p2c4?file=/src/App.js) ⬇️

```css
.css-7wnhq1 {
  color: hotpink;
}
.css-7wnhq1:hover {
  color:  red;
}
```

**Comment by Andarist** [comment]
@oliviertassinari yeah, I plan to remove this code in the next major version. It was only introduced to ease the transition from Emotion 10 to Emotion 11.

For the time being, we probably gonna stick to LESS/SASS semantics. They are still vastly popular and this is what people got used to. This doesn't conflict with the CSS nesting proposal - it's just an additional syntax that would be allowed.

If you find this `compat` plugin to be problematic with some of the things that you are doing in MUI - let me know, I could try to figure out what can be done about it in Emotion 11 without breaking changes.
